community/kernel/src/test/java/org/neo4j/test/TestSecurityManager.java

```diff
@@ -0,0 +1,440 @@
+        }
+        else
+        {
+            super.checkPropertiesAccess();
+        }
+    }
+
+    @Override
+    public void checkPropertyAccess( String key )
+    {
+        if ( isManagerExist() )
+        {
+            securityManager.checkPropertyAccess( key );
+        }
+        else
+        {
+            super.checkPropertyAccess( key );
+        }
+    }
+
+    @Override
+    public boolean checkTopLevelWindow( Object window )
+    {
+        return isManagerExist() ? securityManager.checkTopLevelWindow( window ) : super.checkTopLevelWindow( window );
+    }
+
+    @Override
+    public void checkPrintJobAccess()
+    {
+        if ( isManagerExist() )
+        {
+            securityManager.checkPrintJobAccess();
+        }
+        else
+        {
+            super.checkPrintJobAccess();
+        }
+    }
+
+    @Override
+    public void checkSystemClipboardAccess()
+    {
+        if ( isManagerExist() )
+        {
+            securityManager.checkSystemClipboardAccess();
+        }
+        else
+        {
+            super.checkSystemClipboardAccess();
+        }
+    }
+
+    @Override
+    public void checkAwtEventQueueAccess()
+    {
+        if ( isManagerExist() )
+        {
+            securityManager.checkAwtEventQueueAccess();
+        }
+        else
+        {
+            super.checkAwtEventQueueAccess();
+        }
+    }
+
+    @Override
+    public void checkPackageAccess( String pkg )
+    {
+        if ( isManagerExist() )
+        {
+            securityManager.checkPackageAccess( pkg );
+        }
+        else
+        {
+            super.checkPackageAccess( pkg );
+        }
+    }
+
+    @Override
+    public void checkPackageDefinition( String pkg )
+    {
+        if ( isManagerExist() )
+        {
+            securityManager.checkPackageDefinition( pkg );
+        }
+        else
+        {
+            super.checkPackageDefinition( pkg );
+        }
+    }
+
+    @Override
+    public void checkSetFactory()
+    {
+        if ( isManagerExist() )
+        {
+            securityManager.checkSetFactory();
+        }
+        else
+        {
+            super.checkSetFactory();
+        }
+    }
+
+    @Override
+    public void checkMemberAccess( Class<?> clazz, int which )
+    {
+        if ( isManagerExist() )
+        {
+            securityManager.checkMemberAccess( clazz, which );
+        }
+        else
+        {
+            super.checkMemberAccess( clazz, which );
+        }
+    }
+
+    @Override
+    public void checkSecurityAccess( String target )
+    {
+        if ( isManagerExist() )
+        {
+            securityManager.checkSecurityAccess( target );
+        }
+        else
+        {
+            super.checkSecurityAccess( target );
+        }
+    }
+
+    @Override
+    public ThreadGroup getThreadGroup()
+    {
+        return isManagerExist() ? securityManager.getThreadGroup() : super.getThreadGroup();
+    }
+
+    private boolean isManagerExist()
```

Comment:
this method name feels really strange to me. how about just calling it `managerExists()` instead?